shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionFeature.kt

```diff
@@ -79,15 +79,29 @@ interface StepCompletionFeature {
         /**
          * Show problem of day solve modal
          */
-        data class ProblemOfDaySolved(val earnedGemsText: String) : Message
+        data class ProblemOfDaySolved(
+            val earnedGemsText: String,
+            val streakText: String?,
+            val streak: Int?
```

Comment:
I think in such cases it's better to use sealed interface, there is implicit dependency that `streakText` and `streak` should be both null or not. With sealed interface we can make this dependency explicit:
```
sealed interface StreakData {
    object Empty : StreakData
    data class Content(val streakText: String, val streak: Int)
}
```